src/adagrams.js

```diff
@@ -1,15 +1,143 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterPool = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
+  
+  
+
+  const letters = Object.keys(letterPool);
+  let hand = [];
+  while (hand.length < 10) {
+    let letter = letters[Math.floor(Math.random() * letters.length)];
+    letterPool[letter] = letterPool[letter] - 1;
+    if (letterPool[letter] > 0) {
+      hand.push(letter);
+    };
+  };
+  return hand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const testWord = input.toUpperCase();
+  let hand = [];
+
+  for(let letter of lettersInHand) {
+    hand.push(letter);
+  };
+
+  for(let characters of testWord){
+    if (hand.includes(characters)) {
+      hand.splice(characters, 1);
+    } else {
+      return false;
+    };
+  };
+
+  return true
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  const scoreChart= {
```

Comment:
This score chart object could be moved into the main scope so it wouldn't be created each time the function `scoreWord` is called. Also, note the object keys in JavaScript don't need quotes.